compiler/src/main/java/org/qbicc/type/definition/DefinedTypeDefinitionImpl.java

```diff
@@ -192,9 +207,165 @@ public LoadedTypeDefinition load() throws VerifyFailedException {
+        if (isInterface()) {
+            methods = cnt == 0 ? MethodElement.NO_METHODS : new MethodElement[cnt];
+            for (int i = 0; i < cnt; i ++) {
+                methods[i] = methodResolvers[i].resolveMethod(methodIndexes[i], this);
+            }
+        } else {
+            List<MethodElement> methodsList = new ArrayList<>(cnt + (cnt >> 1)); // 1.5x size
+            for (int i = 0; i < cnt; i ++) {
+                methodsList.add(methodResolvers[i].resolveMethod(methodIndexes[i], this));
+            }
+            // now add methods for any maximally-specific interface methods that are not implemented by this class
+            // - but, if the method is default, let's copy it in, body and all
+            HashMap<String, HashSet<MethodDescriptor>> visited = new HashMap<>();
+            // populate the set of methods that we already have implementations for
+            for (MethodElement method : methodsList) {
+                addMethodToVisitedSet(method, visited);
+            }
+            addMethodsToVisitedSet(superType, visited);
+            // now the visited set contains all methods we do not need to implement
+            // next we find the set of methods that we're missing
+            for (int depth = 0;; depth ++) {
+                // at each depth stage, create a new to-add set
+                HashMap<String, HashMap<MethodDescriptor, List<MethodElement>>> toAdd = new HashMap<>();
+                if (findInterfaceImplementations(interfaces, visited, toAdd, depth) == 0) {
+                    break;
+                }
+                // we may have found methods to add
+                for (Map.Entry<String, HashMap<MethodDescriptor, List<MethodElement>>> entry : toAdd.entrySet()) {
+                    String name = entry.getKey();
+                    HashMap<MethodDescriptor, List<MethodElement>> subMap = entry.getValue();
+                    nextMethod: for (Map.Entry<MethodDescriptor, List<MethodElement>> entry2 : subMap.entrySet()) {
+                        MethodDescriptor descriptor = entry2.getKey();
+                        List<MethodElement> elementList = entry2.getValue();
+                        MethodElement defaultMethod = null;
+                        MethodElement oneOfTheMethods = null;
+                        for (MethodElement element : elementList) {
+                            if (oneOfTheMethods == null) {
+                                oneOfTheMethods = element;
+                            }
+                            if (! element.isAbstract()) {
+                                // found a default method
+                                if (defaultMethod == null) {
+                                    defaultMethod = element;
+                                } else {
+                                    // conflict method! Synthesize a method that throws ICCE
+                                    MethodElement.Builder builder = MethodElement.builder();
```

Comment:
I think we're also missing a "most specific" check here.  Finding more than 1 method is OK provided one is more specific than another.

Here's an example where a sub-interface is more specific:
```
interface I { default void m() {...} }
interface J { default void m() {...} }
interface K extends I, J { default void m() {...} }

class C extends I, J, K {}
```

We can handle that here or when building the `List<MethodElement>`.  In either case, we need to check each element in the list to see if it's more specific than existing elements and remove them if they are. 

After processing I and J, our list would have `{ I::m, J::m }` but when we add `K::m`, it's more specific than both and should remove both of them from the list.

This could also be handled in `findInterfaceImplementations`

Reply:
Notice that the loop starting on 231 is a breadth-first search. Methods are detected by specificity, so in your example we'd hit K first, and then `m` would be in the visited set so it would be excluded when we hit `I` and `J`.

Reply:
OK yeah it looks like this case needs to be handled because the interfaces appear at multiple depths.  I just have to work out the algorithmic implication.

Reply:
I think handling it during the list building makes the most sense.

Reply:
When I implemented this in OpenJ9, the potential conflict lists tended to be quite small so I didn't worry about the repeated processing of it.

Reply:
I have pushed a potential fix.

Reply:
Looks good to me